packages/shared/src/hooks/useChat.ts

```diff
@@ -176,13 +81,96 @@ export const useChat = ({ id: idFromProps, query }: UseChatProps): UseChat => {
         sourceRef.current.close();
       }
 
-      setSearchQuery(undefined);
+      setSessionId(undefined);
+
+      let queryKey: ReturnType<typeof generateQueryKey> = null;
```

Comment:
Just to note according to TKdodo this is very error prone and queryKeys should ideally always be consts as close to the query as possible.
I do see why you chose this and just leaving this as a mention, not sure if there even is a action point

Reply:
> Just to note according to TKdodo this is very error prone and queryKeys should ideally always be consts as close to the query as possible.

I know what you are hinting at but the fact we have a generator with `generateQueryKey` makes it ok since we don't generate query keys by hand.